src/bond.ts

```diff
@@ -568,57 +623,264 @@ export class BondContract extends Contract {
     return setContractStateTx;
   }
 
+  /**
+   *Builds a transaction to blacklist addresses.
+   * @param senderAddress - The address of the sender.
+   * @param addresses - An array of addresses to be removed from the blacklist.
+   */
   setAcceptedCallers(senderAddress: IAddress, addresses: IAddress[]) {
-    throw new Error('Not implemented');
+    const inputAddresses = addresses.map(
+      (address) => new AddressValue(address)
+    );
+    const tx = new Transaction({
+      value: 0,
+      data: new ContractCallPayloadBuilder()
+        .setFunction('setBlacklist')
+        .setArgs(inputAddresses)
+        .build(),
+      receiver: this.contract.getAddress(),
+      sender: senderAddress,
+      gasLimit: 20_000_000,
+      chainID: this.chainID
+    });
+    return tx;
   }
 
+  /**
+   *Builds a transaction to blacklist addresses.
+   * @param senderAddress - The address of the sender.
+   * @param addresses - An array of addresses to be removed from the blacklist.
```

Comment:
description not accurate.

```suggestion
   * @param addresses - An array of addresses to be ADDED to the blacklist.
```